examples/sundials/ex16.cpp

```diff
@@ -258,61 +298,90 @@ int main(int argc, char *argv[])
    }
 
    // 7. Define the ODE solver used for time integration.
-   double t = 0.0;
-   ODESolver *ode_solver = NULL;
-   CVODESolver *cvode = NULL;
-   ARKStepSolver *arkode = NULL;
+   real_t t = 0.0;
+   std::unique_ptr<ODESolver> ode_solver;
    switch (ode_solver_type)
    {
       // MFEM explicit methods
-      case 1: ode_solver = new ForwardEulerSolver; break;
-      case 2: ode_solver = new RK2Solver(0.5); break; // midpoint method
-      case 3: ode_solver = new RK3SSPSolver; break;
-      case 4: ode_solver = new RK4Solver; break;
+      case 1: ode_solver = std::make_unique<ForwardEulerSolver>(); break;
```

Comment:
The function template `make_unique` requires C++14 -- newer versions of SUNDIALS also require C++14, so with such versions using `make_unique` is fine. What about older SUNDIALS versions that we still support in MFEM?

Currently, both build systems switch to C++14 if SUNDIALS is enabled regardless of the SUNDIALS version, so using `make_unique` in the SUNDIALS examples will be okay. However, in the MFEM Spack package the logic for switching to C++14 is when using SUNDIALS v6.4.0 or newer, so the use of `make_unique` can create an issue in the corner case where `mfem@develop` (after merging this branch) is built with `^sundials@6.3.0`. I don't know if it's that important to worry about this corner case though.

If you want to be fully consistent, you can switch to `ode_solver.reset(new ForwardEulerSolver)`, however, I'll leave to you to decide if you think this change is worth it.

Reply:
> The function template make_unique requires C++14

Ah right, I keep forgetting to think about what was introduced in C++14.  Good catch.

> What about older SUNDIALS versions that we still support in MFEM?

I did confirm with the SUNDIALS team that C++14 is the only standard requirement.  Sounds like the MFEM Spack package logic could be made consistent with the build systems to switch to C++14 if any version of SUNDIALS is used?

Reply:
It's a corner case, let's not worry about it. If someone runs into it, we can fix it then.